### PR TITLE
[Fix #3944] Allow keyword args in Style/RaiseArgs

### LINKED ISSUE #3944: Style/RaiseArgs doesn't handle exception classes with just keyword arguments properly
## Expected behavior

When using an exception class with just keyword arguments, it should not find an offense.
``` ruby
raise MyError.new(keyword: argument, key: word)
```

## Actual behavior

It finds an offense, telling me I should be using `raise MyError, message` instead. This is because the code simply checks if there's only 1 argument, and keyword argument always count as one.

## Steps to reproduce the problem

Run rubocop on the following snippet, with Style/RaiseArgs enabled

``` ruby
raise MyError.new(keyword: argument, key: word)
```
## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
rubocop -V
0.46.0 (using Parser 2.3.3.1, running on ruby 2.3.3 x86_64-darwin16)
```

### PR DESCRIPTION
**Pull request by mikegee**
Allow code like `raise MyError.new(keyword: argument, kw: arg)` by default.

fixes #3944 

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests are passing.
* [x] The new code doesn't generate RuboCop offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

### CODE REVIEW

**Comment by mikegee** [comment]
Let me know if this is desirable.

**Comment by bbatsov** [comment]
Looks reasonable to me, although the class's documentation should be updated to reflect this as well.

File: spec/rubocop/cop/style/raise_args_spec.rb [hunk 1]
```diff
@@ -142,6 +142,11 @@
       expect(cop.offenses).to be_empty
     end
 
+    it 'accepts exception constructor with keyword arguments' do
+      inspect_source(cop, 'raise RuntimeError.new(a: 1, b: 2)')
```
Comment (bbatsov) [inline spec/rubocop/cop/style/raise_args_spec.rb]: Does `RuntimeError` support this? Not a big deal, but might confuse someone.

File: lib/rubocop/cop/style/raise_args.rb [hunk 2]
```diff
@@ -111,18 +111,27 @@ def check_exploded(node)
...
             opposite_style_detected
           end
         end
 
+        def acceptable_exploded_args?(args)
+          # Allow code like `raise Ex.new(arg1, arg2)`.
+          return true if args.size > 1
+
+          # Disallow zero arguments.
+          return false if args.size.zero?
```
Comment (Drenmi) [inline lib/rubocop/cop/style/raise_args.rb]: `size.zero?` == `empty?` 😀

Comment (bbatsov) [inline lib/rubocop/cop/style/raise_args.rb]: Don't we have a cop for this??? 🗡 😀

Comment (Drenmi) [inline lib/rubocop/cop/style/raise_args.rb]: Good point. I think maybe the cop only checks `size == 0`? 🤔

Comment (bbatsov) [inline lib/rubocop/cop/style/raise_args.rb]: Probably you're right.

**Comment by Drenmi** [comment]
Looks like CI (MRI 2.3.3 and 2.4.0) is failing for all new PRs because of an [issue in `rainbow`](https://github.com/sickill/rainbow/issues/44).

File: spec/rubocop/cop/style/raise_args_spec.rb (hunk 1 above)
Comment (mikegee) [inline spec/rubocop/cop/style/raise_args_spec.rb]: Good point. I'll change the others that give multiple arguments to `RuntimeError` while I'm correcting this.

**Comment by bbatsov** [comment]
That's unfortunate. Hopefully it will be resolved soon.

**Pull request merged**